eng/install-native-dependencies.sh

```diff
@@ -10,26 +10,41 @@ set -e
 
 os="$(echo "$1" | tr "[:upper:]" "[:lower:]")"
 
-if [ -e /etc/os-release ]; then
-    . /etc/os-release
+if [ -z "$os" ]; then
+    . "$(dirname "$0")"/native/init-os-and-arch.sh
 fi
 
-if [ "$os" = "linux" ] && { [ "$ID" = "debian" ] || [ "$ID_LIKE" = "debian" ]; }; then
-    apt update
+case "$os" in
+    linux)
+        if [ -e /etc/os-release ]; then
+            . /etc/os-release
+        fi
 
-    apt install -y build-essential gettext locales cmake llvm clang lldb liblldb-dev libunwind8-dev libicu-dev liblttng-ust-dev \
-        libssl-dev libkrb5-dev libnuma-dev zlib1g-dev
+        if [ "$ID" != "debian" ] && [ "$ID_LIKE" != "debian" ]; then
+            echo "Unsupported distro. distro: $ID"
+            exit 1
+        fi
 
-    localedef -i en_US -c -f UTF-8 -A /usr/share/locale/locale.alias en_US.UTF-8
-elif [ "$os" = "maccatalyst" ] || [ "$os" = "osx" ] || [ "$os" = "macos" ] || [ "$os" = "tvos" ] || [ "$os" = "ios" ]; then
-    echo "Installed xcode version: $(xcode-select -p)"
+        apt update
 
-    export HOMEBREW_NO_INSTALL_CLEANUP=1
-    export HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK=1
-    # Skip brew update for now, see https://github.com/actions/setup-python/issues/577
-    # brew update --preinstall
-    brew bundle --no-upgrade --no-lock --file "$(dirname "$0")/Brewfile"
-else
-    echo "Must pass 'linux', 'macos', 'maccatalyst', 'ios' or 'tvos' as first argument."
-    exit 1
-fi
+        apt install -y build-essential gettext locales cmake llvm clang lldb liblldb-dev libunwind8-dev libicu-dev liblttng-ust-dev \
+            libssl-dev libkrb5-dev libnuma-dev zlib1g-dev
+
+        localedef -i en_US -c -f UTF-8 -A /usr/share/locale/locale.alias en_US.UTF-8
+        ;;
+
+    osx|mac*|ios*|tvos*)
```

Comment:
nit for a future PR: I prefer listing the OSes explicitly (`maccatalyst`, `ios`, `iossimulator`) as it makes it easier to find matches if you're grepping the codebase

Reply:
I opted for this approach because there was `macos` option in old revision and the rest are no different than `StartsWith('ios/tvos')` per-existing pattern. I agree that we should update such conditions together and bring them to the same plan to keep things plain and grep'able. Will follow-up after https://github.com/dotnet/runtime/pull/82086.

Reply:
I opted for this approach because there was `macos` option in old revision and the rest are no different than `StartsWith('ios/tvos')` per-existing pattern. I agree that we should update such conditions together and bring them to the same plan to keep things plain and grep'able. Will follow-up after https://github.com/dotnet/runtime/pull/82086 is merged.